homeassistant/components/somfy/__init__.py

```diff
@@ -0,0 +1,186 @@
+"""
+Support for Somfy hubs.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/somfy/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import callback
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+API = 'api'
+
+DEVICES = 'devices'
+
+REQUIREMENTS = ['pymfy==0.4.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+DOMAIN = 'somfy'
+
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+
+NOTIFICATION_CB_ID = 'somfy_cb_notification'
+NOTIFICATION_OK_ID = 'somfy_ok_notification'
+NOTIFICATION_TITLE = 'Somfy Setup'
+
+ATTR_ACCESS_TOKEN = 'access_token'
+ATTR_REFRESH_TOKEN = 'refresh_token'
+ATTR_CLIENT_ID = 'client_id'
+ATTR_CLIENT_SECRET = 'client_secret'
+
+SOMFY_AUTH_CALLBACK_PATH = '/auth/somfy/callback'
+SOMFY_AUTH_START = '/auth/somfy'
+
+DEFAULT_CACHE_PATH = '.somfy'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SOMFY_COMPONENTS = ['cover']
+
+
+def setup(hass, config):
+    """Set up the Somfy component."""
+    from pymfy.api.somfy_api import SomfyApi
+
+    hass.data[DOMAIN] = {}
+    is_ready = False
+
+    # This is called to create the redirect so the user can Authorize Home .
+    redirect_uri = '{}{}'.format(
+        hass.config.api.base_url, SOMFY_AUTH_CALLBACK_PATH)
+    conf = config[DOMAIN]
+    api = SomfyApi(conf.get(CONF_CLIENT_ID),
+                   conf.get(CONF_CLIENT_SECRET),
+                   redirect_uri, hass.config.path(DEFAULT_CACHE_PATH))
+    hass.data[DOMAIN][API] = api
+
+    if not api.token:
```

Comment:
I would look into building https://developers.home-assistant.io/docs/en/config_entries_index.html

Reply:
Yes, I've already look at this. But I don't see how I can use it here. As said in my upper comment, all I need is a token. This token is automatically retrieved when the end user click on the link within the notification I displayed. There is no information asked, except the one to set in `configuration.yaml`. The token is then given to [pymfy](https://pypi.org/project/pymfy/) storing it as a file. This way, pymfy know if the use is authenticated or not.

Reply:
is that what the cache is for ?

Reply:
Yes.